messenger/multiple_buses.rst

```diff
@@ -0,0 +1,288 @@
+                <framework:messenger default-bus="messenger.bus.commands">
+                    <framework:bus name="messenger.bus.commands">
+                        <framework:middleware id="validation" />
+                        <framework:middleware id="doctrine_transaction" />
+                    <framework:bus>
+                    <framework:bus name="messenger.bus.queries">
+                        <framework:middleware id="validation" />
+                    <framework:bus>
+                    <framework:bus name="messenger.bus.events">
+                        <framework:middleware id="validation" />
+                        <framework:middleware id="allow_no_handler" />
+                    <framework:bus>
+                </framework:messenger>
+            </framework:config>
+        </container>
+
+    .. code-block:: php
+
+        // config/packages/messenger.php
+        $container->loadFromExtension('framework', array(
+            'messenger' => array(
+                // The bus that is going to be injected when injecting MessageBusInterface
+                'default_bus' => 'messenger.bus.commands',
+                'buses' => array(
+                    'messenger.bus.commands' => array(
+                        'middleware' => array(
+                            'validation',
+                            'doctrine_transaction',
+                        ),
+                    ),
+                    'messenger.bus.queries' => array(
+                        'middleware' => array(
+                            'validation',
+                        ),
+                    ),
+                    'messenger.bus.events' => array(
+                        'middleware' => array(
+                            'validation',
+                            'allow_no_handler',
+                        ),
+                    ),
+                ),
+            ),
+        ));
+
+This will generate the ``messenger.bus.commands``, ``messenger.bus.queries``
+and ``messenger.bus.events`` services that you can inject in your services.
+
+Type-hints and Auto-wiring
+--------------------------
+
+Auto-wiring is a great feature that allows you to reduce the amount of configuration
+required for your service container to be created. When using multiple buses, by default,
+the auto-wiring will not work as it won't know which bus to inject in your own services.
+
+In order to clarify this, you can use the DependencyInjection's binding capabilities
+to clarify which bus will be injected based on the argument's name:
+
+.. configuration-block::
+
+    .. code-block:: yaml
+
+        # config/services.yaml
+        services:
+            _defaults:
+                # ...
+
+                bind:
+                    $commandBus: '@messenger.bus.commands'
+                    $queryBus: '@messenger.bus.queries'
+                    $eventBus: '@messenger.bus.events'
```

Comment:
Just a note - we'll need another PR in Symfony 4.2, because (I believe) these bindings are done automatically.